server/stycoon/command_trade.go

```diff
@@ -0,0 +1,24 @@
+package stycoon
+
+import (
+	"github.com/gdg-garage/space-tycoon/server/database"
+)
+
+func NewTradeCommand(command Command) *TradeCommand {
+	return &TradeCommand{
+		Type:     command.Type,
+		Amount:   command.Amount,
+		Resource: command.Resource,
+		Target:   command.Target,
+	}
+}
+
+func (game *Game) processTrade(id int64, command Command) error {
+	c := NewTradeCommand(command)
+	err := AssertTradeCommandRequired(*c)
+	if err != nil {
+		return err
+	}
+	// TODO should we validate trading with own ships?
```

Comment:
the database could theoretically validate that. but that is too late to report a meaningful error. therefore you should validate that.

Reply:
done